test/stdlib/ErrorBridged.swift

```diff
@@ -722,4 +722,38 @@ ErrorBridgingTests.test("Error archetype identity") {
     === nsError)
 }
 
+// SR-9389
+@objc(SR9389Error) enum Rdar46420910Error: Int, Error {
+  case wrongDomain
+}
+class ParentA: NSObject {
+  @objc(ParentAError) enum Error: Int, Swift.Error {
+    case failed
+  }
+}
+class ParentB: NSObject {
+  @objc(ParentAError) enum Error: Int, Swift.Error {
+    case failed
+  }
+}
+
+ErrorBridgingTests.test("@objc enum error domains") {
+  // Should contain module and @objc name.
+  expectTrue(Rdar46420910Error.wrongDomain._domain.contains("main"))
+  expectTrue(Rdar46420910Error.wrongDomain._domain.contains("SR9389Error"))
+
+  // Should not contain Swift name.
+  expectFalse(Rdar46420910Error._nsErrorDomain.contains("Rdar46420910Error"))
+
+  // Should contain domain constant name (common Obj-C convention).
+  expectTrue(Rdar46420910Error._nsErrorDomain.contains("SR9389ErrorDomain"))
+
+  // Identically-named children of different parents should not be castable to
+  // one another.
+  expectTrue(ParentA.Error.failed as NSError is ParentA.Error)
```

Comment:
I just realized that these last four tests are written in a form that wouldn't fail in master. I'll fix it tomorrow.